src/GroupManager.php

```diff
@@ -77,12 +77,12 @@ public function getGroupsForEntityType($entity_type_id) {
    *
    * @return array
    */
-  public function getAllGroups() {
+  public function GetAllGroupBundles($entity_type = NULL) {
     if (!isset($this->groupMap)) {
       $this->refreshGroupMap();
     }
 
-    return $this->groupMap;
+    return !empty($this->groupMap[$entity_type]) ? $this->groupMap[$entity_type] : $this->groupMap;
```

Comment:
This is not right either, as if an entity type that has no group data is passed in, you will get all group data.